test/powershell/engine/Api/TypeInference.Tests.ps1

```diff
@@ -504,7 +504,7 @@ Describe "Type inference Tests" -tags "CI" {
     It "Infers typeof Select-Object when Parameter is ExcludeProperty" {
         $res = [AstTypeInference]::InferTypeOf( {  [io.fileinfo]::new("file")  |  Select-Object -ExcludeProperty *Time*, E* }.Ast)
         $res.Count | Should -Be 1
-        $res[0].Name | Should -BeExactly "System.Management.Automation.PSObject#Attributes:BaseName:Directory:DirectoryName:FullName:IsReadOnly:Length:LengthString:LinkTarget:LinkType:Mode:ModeWithoutHardLink:Name:NameString:ResolvedTarget:Target:VersionInfo"
+        $res[0].Name | Should -BeExactly "System.Management.Automation.PSObject#Attributes:BaseName:Directory:DirectoryName:FullName:IsReadOnly:Length:LengthString:LinkTarget:LinkType:Mode:ModeWithoutHardLink:Name:NameString:ResolvedTarget:Target:UnixFileMode:VersionInfo"
```

Comment:
PowerShell has a `UnixMode` script property for `FileInfo` and `DirectoryInfo` on Unix platforms. I wonder if the new `UnixFileMode` would bring any issues for the existing script property ... like what if they are showing different results?
Can you check what is returned by `UnixFileMode` on Linux?

Reply:
This is how it looks:

```
PS /mnt/d/PSGit/PowerShell> Get-Item ./build.psm1 | fl UnixMode, UnixFileMode

UnixMode     : -rwxrwxrwx
UnixFileMode : OtherExecute, OtherWrite, OtherRead, GroupExecute, GroupWrite, GroupRead, UserExecute, UserWrite, UserRe
               ad

```

Reply:
Also, this:

```
PS /mnt/d/PSGit/PowerShell> Get-Item ./a.txt | fl UnixMode, UnixFileMode

UnixMode     : -r-xr-xr-x
UnixFileMode : OtherExecute, OtherRead, GroupExecute, GroupRead, UserExecute, UserRead
```

They seems to be same. We may want to consider removing the script property. That is a different issue though

Reply:
Looks good!